util/src/main/java/org/datacommons/util/SummaryReportGenerator.java

```diff
@@ -70,10 +85,51 @@ public int getNumObservations() {
     public List<Double> getSeriesValues() {
       return this.seriesValues;
     }
-  }
 
-  public static boolean TEST_mode = false;
-  public static final String SUMMARY_REPORT_HTML = "summary_report.html";
+    public String getTimeSeriesChartSVG() {
+      TimeSeriesCollection dataset = new TimeSeriesCollection();
+      TimeSeries timeSeries = new TimeSeries("ts");
+      for (int i = 0; i < this.seriesDates.size(); i++) {
+        if (i >= this.seriesValues.size()) break;
+        LocalDateTime localDateTime = StringUtil.getValidISO8601Date(seriesDates.get(i));
+        if (localDateTime == null) continue;
+        Date date = Date.from(localDateTime.atZone(ZoneId.systemDefault()).toInstant());
+        timeSeries.addOrUpdate(new Day(date), this.seriesValues.get(i));
+      }
+      if (timeSeries.isEmpty()) return "";
+      dataset.addSeries(timeSeries);
+      JFreeChart chart = ChartFactory.createTimeSeriesChart("", "", "", dataset);
+      XYItemRenderer renderer = new XYLineAndShapeRenderer(true, true);
+      renderer.setSeriesShape(0, ShapeUtils.createDiamond(5));
+      renderer.setSeriesVisibleInLegend(0, false);
+      XYPlot plot = chart.getXYPlot();
+      plot.setBackgroundPaint(Color.WHITE);
+      plot.setRenderer(renderer);
+      if (timeSeries.findValueRange().getLength() == 0) {
+        // Manually set the range when the values in the time series are all the same. Otherwise,
+        // the chart library will draw an axis with multiple ticks all labeled with that same value.
+        plot.getRangeAxis().setRange(0, this.seriesValues.get(0) * 2);
+      }
+      if (this.seriesDates.get(0).equals(this.seriesDates.get(this.seriesDates.size() - 1))) {
+        // Override the date formatter for the x axis when there is only one data point. Otherwise,
+        // the chart library will label the single date as 00:00:00.
+        DateAxis xAxis = (DateAxis) plot.getDomainAxis();
+        String datePattern = StringUtil.getValidISO8601DatePattern(this.seriesDates.get(0));
+        if (datePattern.isEmpty() || StringUtil.EXTRA_DATE_PATTERNS.contains(datePattern)) {
```

Comment:
nit: should we have a helper in StringUtil like getValidISO8601DateTemplate() that takes a pattern returns the template ("yyyy-MM-dd" type strings)?

Reply:
getValidISO8601DatePattern returns the template, but it just also returns the [2 extra date patterns](https://github.com/datacommonsorg/import/blob/7b538e9aeaddd060ebb2db75e55459530a027b6a/util/src/main/java/org/datacommons/util/StringUtil.java#L50) that can't be parsed in Java that are considered valid in our CPP implementation. I feel like adding another helper like getValidISO8601DateTemplate might not be needed?

Reply:
EXTRA_DATE_PATTERNS is sort of an implementation detail, so I thought it might be cleaner to have an API that wraps the EXTRA_DATE_PATTERNS check and only returns a template.  But its fine...

Reply:
I see what you mean, done